docker/lib/dependabot/docker/file_fetcher.rb

```diff
@@ -40,63 +42,76 @@ def fetch_files
         elsif incorrectly_encoded_dockerfiles.none?
           raise(
             Dependabot::DependencyFileNotParseable,
-            incorrectly_encoded_yamlfiles.first.path
+            T.must(incorrectly_encoded_yamlfiles.first).path
           )
         else
           raise(
             Dependabot::DependencyFileNotParseable,
-            incorrectly_encoded_dockerfiles.first.path
+            T.must(incorrectly_encoded_dockerfiles.first).path
           )
         end
       end
 
       private
 
+      sig { returns(T::Array[DependencyFile]) }
       def dockerfiles
-        @dockerfiles ||=
-          repo_contents(raise_errors: false)
+        @dockerfiles ||= T.let(fetch_dockerfiles, T.nilable(T::Array[DependencyFile]))
+      end
+
+      sig { returns(T::Array[Dependabot::DependencyFile]) }
+      def fetch_dockerfiles
+        repo_contents(raise_errors: false)
           .select { |f| f.type == "file" && f.name.match?(DOCKER_REGEXP) }
           .map { |f| fetch_file_from_host(f.name) }
       end
 
+      sig { returns(T::Array[DependencyFile]) }
       def correctly_encoded_dockerfiles
-        dockerfiles.select { |f| f.content.valid_encoding? }
+        dockerfiles.select { |f| f.content&.valid_encoding? }
       end
 
+      sig { returns(T::Array[DependencyFile]) }
       def incorrectly_encoded_dockerfiles
-        dockerfiles.reject { |f| f.content.valid_encoding? }
+        dockerfiles.reject { |f| f.content&.valid_encoding? }
       end
 
+      sig { returns(T::Array[DependencyFile]) }
```

Comment:
should this be `nilable` too?

Reply:
No, since we `T.let(` the expression that we return. It reads a little weird, because the `@yamlfiles` variable _is_ nilable, since if the method is not called, it is nil, but then the value returned from `repo_contents(raise_errors: false).select ....` always returns a value, and so by the time the method returns we know it is not `nil`, even though `@yamlfiles` _is_ `T.nilable(T::Array[DependencyFile])`.